schemes/musica/musica_ccpp_util.F90

```diff
@@ -37,4 +42,53 @@ logical function has_error_occurred(error, error_message, error_code)
 
   end function has_error_occurred
 
+  !> Calculate the solar zenith angle and Earth-Sun distance
+  !> @param[in] calendar_day Calendar day, including fraction
+  !> @param[in] latitude Latitude in radians
+  !> @param[in] longitude Longitude in radians
+  !> @param[in] earth_eccentricity Earth's eccentricity factor (unitless)
+  !> @param[in] earth_obliquity Earth's obliquity in radians
+  !> @param[in] perihelion_longitude Earth's mean perihelion longitude at the vernal equinox (radians)
+  !> @param[in] moving_vernal_equinox_longitude Earth's moving vernal equinox longitude of perihelion plus pi (radians)
+  !> @param[out] solar_zenith_angle Solar zenith angle in radians
+  !> @param[out] earth_sun_distance Earth-Sun distance in AU
+  !> @param[out] errmsg Error message
+  !> @param[out] errcode Error code
+  subroutine calculate_solar_zenith_angle_and_earth_sun_distance(calendar_day, &
+      latitude, longitude, earth_eccentricity, earth_obliquity, perihelion_longitude, &
+      moving_vernal_equinox_longitude, solar_zenith_angle, earth_sun_distance, &
+      errmsg, errcode)
+    use shr_orb_mod, only: shr_orb_decl, shr_orb_cosz
+    use musica_util, only: error_t
+
+    real(kind_phys),    intent(in)  :: calendar_day
+    real(kind_phys),    intent(in)  :: latitude(:)
+    real(kind_phys),    intent(in)  :: longitude(:)
+    real(kind_phys),    intent(in)  :: earth_eccentricity
+    real(kind_phys),    intent(in)  :: earth_obliquity
+    real(kind_phys),    intent(in)  :: perihelion_longitude
+    real(kind_phys),    intent(in)  :: moving_vernal_equinox_longitude
+    real(kind_phys),    intent(out) :: solar_zenith_angle(:)
+    real(kind_phys),    intent(out) :: earth_sun_distance
+    character(len=512), intent(out) :: errmsg
+    integer,            intent(out) :: errcode
+
+    real(kind_phys) :: delta
+    integer :: i_sza
+
+    errcode = 0
+    errmsg = ''
+
+    ! Calculate earth/orbit parameters
+    call shr_orb_decl(calendar_day, earth_eccentricity, earth_obliquity, &
+                      perihelion_longitude, moving_vernal_equinox_longitude, &
+                      delta, earth_sun_distance)
```

Comment:
This function is imported, but if we or CAM-SIMA are going to rewrite it, is there a chance we could rename the function?

Reply:
I think the plan now is to use this function in CAM-SIMA and pass the inputs needed for TUV-x through CCPP, so I will remove this copy of these shared functions. (But, I totally agree with renaming this - it just might be that this function is used in a lot of places in CESM, so it might take some discussion first)